airbyte-integrations/connectors/source-lever-hiring/source_lever_hiring/source.py

```diff
@@ -6,18 +6,22 @@
 
 from airbyte_cdk.sources import AbstractSource
 from airbyte_cdk.sources.streams import Stream
-from airbyte_cdk.sources.streams.http.auth import Oauth2Authenticator
+from airbyte_cdk.sources.streams.http.auth import BasicHttpAuthenticator, Oauth2Authenticator
 
 from .streams import Applications, Interviews, Notes, Offers, Opportunities, Referrals, Users
 
 
 def _auth_from_config(config):
-    return Oauth2Authenticator(
-        client_id=config["credentials"]["client_id"],
-        client_secret=config["credentials"]["client_secret"],
-        refresh_token=config["credentials"]["refresh_token"],
-        token_refresh_endpoint=f"{SourceLeverHiring.URL_MAP_ACCORDING_ENVIRONMENT[config['environment']]['login']}oauth/token",
-    )
+
+    if config and config["credemtials"] and 'api_key' in config["credentials"]:
+        return BasicHttpAuthenticator(username=config["credentials"]["api_key"], auth_method="Basic")
+    else:
```

Comment:
https://github.com/airbytehq/airbyte/blob/e1983114bcb3b524c64c6c1f3f74a52da14ba363/airbyte-integrations/connectors/source-lever-hiring/unit_tests/test_source.py#L11-L14

Similar to the unit test for `check_connection` above, can you add one for the basic auth? Thanks.